### PR TITLE
Factorial

### PR DESCRIPTION
**Pull request by zamonier**

### CODE REVIEW

File: recursion/src/main/java/FactorialImpl.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+public class FactorialImpl implements Factorial {
+
+  @Override
+  public int of(int base) {
+    return base != 0
```
Comment (PaulFrmBrn) [inline recursion/src/main/java/FactorialImpl.java]: And negative `base`?

Comment (zamonier) [reply recursion/src/main/java/FactorialImpl.java]: Tests are green)

Comment (PaulFrmBrn) [reply recursion/src/main/java/FactorialImpl.java]: ha! ) It's mirrored conversation in https://github.com/PaulFrmBrn/udemy-ds-and-algo-course/pull/6 )

Comment (zamonier) [reply recursion/src/main/java/FactorialImpl.java]: -7?
Can't get you.
I think we need to update tests - and after rebase I will add checks

Comment (PaulFrmBrn) [reply recursion/src/main/java/FactorialImpl.java]: Ok, I'll do it

Comment (PaulFrmBrn) [reply recursion/src/main/java/FactorialImpl.java]: https://github.com/PaulFrmBrn/udemy-ds-and-algo-course/pull/7

**Pull request merged**